warehouse/query-core/src/main/java/datawave/query/tables/RemoteEdgeQueryLogic.java

```diff
@@ -0,0 +1,269 @@
+        return getConfig();
+    }
+
+    @Override
+    public String getPlan(AccumuloClient connection, Query settings, Set<Authorizations> auths, boolean expandFields, boolean expandValues) throws Exception {
+        GenericResponse<String> planResponse = remoteQueryService.planQuery(getRemoteQueryLogic(), settings.toMap(), getCallerObject());
+        return planResponse.getResult();
+    }
+
+    @Override
+    public void setupQuery(GenericQueryConfiguration genericConfig) throws Exception {
+        if (!RemoteQueryConfiguration.class.isAssignableFrom(genericConfig.getClass())) {
+            throw new QueryException("Did not receive a RemoteQueryConfiguration instance!!");
+        }
+
+        config = (RemoteQueryConfiguration) genericConfig;
+        // Create an iterator that returns a stream of EdgeBase objects
+        iterator = new RemoteQueryLogicIterator();
+    }
+
+    @Override
+    public QueryLogicTransformer getTransformer(Query settings) {
+        // a transformer that turns EdgeBase objects into a response
+        if (transformerInstance == null) {
+            transformerInstance = new EdgeBaseTransformer(settings, getMarkingFunctions(), getResponseObjectFactory());
+        }
+
+        return transformerInstance;
+    }
+
+    @Override
+    public RemoteEdgeQueryLogic clone() {
+        return new RemoteEdgeQueryLogic(this);
+    }
+
+    @Override
+    public void close() {
+
+        super.close();
+
+        log.debug("Closing RemoteQueryLogic: " + System.identityHashCode(this));
+
+        if (getRemoteId() != null) {
+            try {
+                remoteQueryService.close(getRemoteId(), getCallerObject());
+            } catch (Exception e) {
+                log.error("Failed to close remote query", e);
+            }
+        }
+    }
+
+    @Override
+    public RemoteQueryConfiguration getConfig() {
+        if (config == null) {
+            config = RemoteQueryConfiguration.create();
+        }
+
+        return config;
+    }
+
+    public void setConfig(RemoteQueryConfiguration config) {
+        this.config = config;
+    }
+
+    public RemoteQueryService getRemoteQueryService() {
+        return remoteQueryService;
+    }
+
+    public void setRemoteQueryService(RemoteQueryService remoteQueryService) {
+        this.remoteQueryService = remoteQueryService;
+    }
+
+    @Override
+    public AccumuloConnectionFactory.Priority getConnectionPriority() {
+        return AccumuloConnectionFactory.Priority.NORMAL;
+    }
+
+    @Override
+    public Set<String> getOptionalQueryParameters() {
+        return new ShardQueryLogic().getOptionalQueryParameters();
+    }
+
+    @Override
+    public Set<String> getRequiredQueryParameters() {
+        return new ShardQueryLogic().getRequiredQueryParameters();
+    }
+
+    @Override
+    public Set<String> getExampleQueries() {
+        return new ShardQueryLogic().getExampleQueries();
+    }
```

Comment:
The standard EdgeQueryLogic implements both `getOptionalQueryParameters` and `getRequiredQueryParameters`, so I think these overrides could cause inconsistent behavior.

While the standard EdgeQueryLogic does not implement `getExampleQueries` in any meaningful way there may well be other extended query logics that do...personally I would like to see that method removed from the QueryLogic interface altogether, but that's a discussion for another time.

Reply:
good catch.  This was a mistake on my part.  Agreed on the second part